src/main/java/edu/byu/cs/dataAccess/sql/SubmissionSqlDao.java

```diff
@@ -12,69 +14,63 @@
+            new ColumnDefinition<Submission>("net_id", Submission::netId),
+            new ColumnDefinition<Submission>("repo_url", Submission::repoUrl),
+            new ColumnDefinition<Submission>("timestamp", s -> Timestamp.from(s.timestamp())),
+            new ColumnDefinition<Submission>("phase", s -> s.phase().toString()),
+            new ColumnDefinition<Submission>("passed", Submission::passed),
+            new ColumnDefinition<Submission>("score", Submission::score),
+            new ColumnDefinition<Submission>("num_commits", Submission::numCommits),
+            new ColumnDefinition<Submission>("head_hash", Submission::headHash),
+            new ColumnDefinition<Submission>("notes", Submission::notes),
+            new ColumnDefinition<Submission>("rubric", s -> new Gson().toJson(s.rubric())),
+            new ColumnDefinition<Submission>("admin", Submission::admin)
+    };
+    private static Submission readSubmission(ResultSet rs) throws SQLException {
+        String netId = rs.getString("net_id");
+        String repoUrl = rs.getString("repo_url");
+        String headHash = rs.getString("head_hash");
+        Instant timestamp = rs.getTimestamp("timestamp").toInstant();
+        Phase phase = Phase.valueOf(rs.getString("phase"));
+        Boolean passed = rs.getBoolean("passed");
+        float score = rs.getFloat("score");
+        Integer numCommits = rs.getInt("num_commits");
+        String notes = rs.getString("notes");
+        Rubric rubric = new Gson().fromJson(rs.getString("rubric"), Rubric.class);
+        Boolean admin = rs.getBoolean("admin");
+
+        return new Submission(netId, repoUrl, headHash, timestamp, phase, passed, score, numCommits, notes, rubric, admin);
+    }
+
+    private final SqlReader<Submission> sqlReader = new SqlReader<Submission>(
+            "submission", COLUMN_DEFINITIONS, SubmissionSqlDao::readSubmission);
+
     @Override
     public void insertSubmission(Submission submission) {
-        try (var connection = SqlDb.getConnection();
-            PreparedStatement statement = connection.prepareStatement(
-                    """
-                    INSERT INTO submission (net_id, repo_url, timestamp, phase, passed, score, head_hash, num_commits, notes, results, rubric, admin)
-                    VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
-                    """)) {
-            statement.setString(1, submission.netId());
-            statement.setString(2, submission.repoUrl());
-            statement.setTimestamp(3, Timestamp.from(submission.timestamp()));
-            statement.setString(4, submission.phase().toString());
-            statement.setBoolean(5, submission.passed());
-            statement.setFloat(6, submission.score());
-            statement.setString(7, submission.headHash());
-            statement.setInt(8, submission.numCommits());
-            statement.setString(9, submission.notes());
-            statement.setString(10, "{}");
```

Comment:
Note that this removes some references of `Submission.Results` though it doesn't fully deprecate the field. See #230